### ISSUE TITLE
Create RTMP Input

### ISSUE THREAD

**Issue by Martin1982** [Martin1982/live-broadcast-bundle#8]
To stream from software like Open Broadcast Software we need an Input for RTMP. "Why?" you may ask; because 1 Input can stream through multiple outputs from a server with high bandwidth, while still only 1 OBS stream needs to be sent from the live filming location.

**Comment by Martin1982** [comment]
Interesting read for a test-setup; https://obsproject.com/forum/resources/how-to-set-up-your-own-private-rtmp-server-using-nginx.50/

**Comment by relaxnow** [comment]
So I looked into this and I think it is important to note that what is required is an actual server that is up and running for OBS to connect to to send it's data.
While this is not really PHPs thing I found a slightly crufty library that I think can be reused: https://github.com/qwantix/php-rtmp-client (on packagist here: https://packagist.org/packages/qwantix/php-rtmp-client )

I would propose:
1. Refactor the client library to allow being a server. Looking at the source code it looks like most of it is there but I'm not THAT familiar with the RMTP protocol.
2. It should be 2 types of servers, first a 'ReceivingServer', this server should:
2.1. Open a port and wait for a connection.
2.2. Accept an incoming CONNECT
2.3. Receive the packets and put it on a fixed size queue (APC shared memory? Memcache? Redis? Something fast...).
3. Second a 'SendingServer' this server should:
3.1. Open a port and wait for a connection.
3.2. Accept an incoming CONNECT.
3.3. Read the packets from the queue and send them to the client.

In fact I would propose this would be a separate project.
Then all the bundle would have to do is use FFMPEG to connect to the SendingServer on localhost with something like this: https://gist.github.com/bnerd/4069082

**Comment by relaxnow** [comment]
Honestly I would first see if you can get Nginx to do this as PHP is not the greatest with running as a daemon.

**Comment by relaxnow** [comment]
```
  application mypull {
            live on;

            # Pull all streams from remote machine
            # and play locally
            pull rtmp://rtmp3.example.com pageUrl=www.example.com/index.html;
        }
```
Something like this should do the trick.

**Comment by Martin1982** [comment]
Agreed that PHP isn't the right tool, in the feature branch I'm now investigating if we can pre-install and configure a Red5 Server. This server is able to run as a standalone package.

I still feel there is a tradeoff though; installing a non-php dependency to make setup as easy as possible or make the developer set up an RTMP server to keep full control.
In the end I think it should be optional to include Red5, but we'll get to that once the branch progresses.

**Comment by Martin1982** [comment]
First release should be without an implemented RTMP server, there are enough servers out there we can use

**Comment by Martin1982** [comment]
Fixed in dd996512fff3407eb3f5950b21d6150a1225b798
